docs/topics/proc_starting-your-application-on-openshift-in-debugging-mode.adoc

```diff
@@ -1,13 +1,27 @@
+// This is a parameterized module. Parameters used:
+//
+//   springboot-debug: Signifies whether this is used by spring boot or not. 
+//
+// Rationale: This procedure is identical in Swarm and Vert.x, and varies slightly in spring boot.
```

Comment:
This is just not true and whoever told you so was wrong. All of Swarm, Vert.x and Spring Boot use the _same_ OpenJDK image, and so can use the exact same instructions.

Snippet from `/opt/run-java/debug-options`:

```bash
  if [ "x${JAVA_ENABLE_DEBUG}" != "x" -o "x${JAVA_DEBUG_ENABLE}" != "x" -o "x${JAVA_DEBUG}" != "x" ]; then
    local debug_port=${JAVA_DEBUG_PORT:-5005}
    echo "-agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=${debug_port}"
  fi
```

So it's entirely possible to use `JAVA_DEBUG`, `JAVA_ENABLE_DEBUG` and `JAVA_DEBUG_ENABLE` interchangeably. I just picked `JAVA_DEBUG` because it's shortest, but they are equivalent.

Reply:
As an added comment, maybe we should also document that the default debugging port can be changed by passing a different value to `JAVA_DEBUG_PORT`?

Reply:
I don't think you often need to change the debugger port when running the JVM in a container, but at the same time, I agree this can be useful to know. So I don't really mind, but I certainly wouldn't disagree :-)